src/main/java/com/snowflake/snowpark_java/types/CloudProviderToken.java

```diff
@@ -0,0 +1,26 @@
+package com.snowflake.snowpark_java.types;
+
+/** Custom Snowflake class that provides access to accessKeyId & secretAccessKey & token secret object. */
```

Comment:
Thanks for adding the API support here. Can we add a ref to the original file here as a description that this is a read-only file with the file link snowflakeDB repo.